roles/download/tasks/download_file.yml

```
@@ -103,7 +103,7 @@
     become: "{{ not download_localhost }}"
     until: "'OK' in get_url_result.msg or
       'file already exists' in get_url_result.msg or
-      get_url_result.status_code == 304"
+      (get_url_result is defined and get_url_result.status_code == 304)"
```

Since `status_code` can be undefined this fix might not work either. Checking that `get_url_result` is defined won't ensure that `status_code` is defined too.

```suggestion
      get_url_result.status_code | default () == 304"
```
Setting an empty default value for `get_url_result.status_code` should be enough to ensure this task fails but cathing the error when `status_code` is not defined.

---

I did all my test with `get_url_result.status_code is defined`, It seams I forgot it in my commit -_-.
I like your idea, with `default()`.